src/hotspot/cpu/x86/macroAssembler_x86.cpp

```diff
@@ -5611,7 +5613,8 @@ void  MacroAssembler::decode_and_move_klass_not_null(Register dst, Register src)
     movl(dst, src);
   } else {
     if (CompressedKlassPointers::base() != nullptr) {
-      mov64(dst, (int64_t)CompressedKlassPointers::base());
+      // Uses 32-bit mov if base is small enough
+      movptr(dst, (intptr_t)CompressedKlassPointers::base());
     } else {
       xorq(dst, dst);
     }
```

Comment:
This can also be 1 instruction when the base is smaller than $2^{31}$.

Reply:
See my remark above.